openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+
+        # Look through all of this Tally's Filters for the type requested
+        for test_filter in self.filters:
+            if type(test_filter) is filter_type:
+                filter_found = test_filter
+                break
+
+            # Also check to see if the desired filter is wrapped up in an
+            # aggregate
+            elif isinstance(test_filter, openmc.AggregateFilter):
+                if isinstance(test_filter.aggregate_filter, filter_type):
+                    filter_found = test_filter
+                    break
+
+        # If we did not find the Filter, throw an Exception
+        if filter_found is None:
+            msg = 'Unable to find filter type "{0}" in ' \
+                  'Tally ID="{1}"'.format(filter_type, self.id)
+            raise ValueError(msg)
+
+        return filter_found
+
+    def get_nuclide_index(self, nuclide):
+        """Returns the index in the Tally's results array for a Nuclide bin
+
+        Parameters
+        ----------
+        nuclide : str
+            The name of the Nuclide (e.g., 'H1', 'U238')
+
+        Returns
+        -------
+        nuclide_index : int
+            The index in the Tally data array for this nuclide.
+
+        Raises
+        ------
+        KeyError
+            When the argument passed to the 'nuclide' parameter cannot be found
+            in the Tally.
+
+        """
+
+        nuclide_index = -1
+
+        # Look for the user-requested nuclide in all of the Tally's Nuclides
+        for i, test_nuclide in enumerate(self.nuclides):
+
+            # If the Summary was linked, then values are Nuclide objects
+            if isinstance(test_nuclide, openmc.Nuclide):
+                if test_nuclide.name == nuclide:
+                    nuclide_index = i
+                    break
+
+            # If the Summary has not been linked, then values are ZAIDs
+            else:
+                if test_nuclide == nuclide:
+                    nuclide_index = i
+                    break
+
+        if nuclide_index == -1:
+            msg = 'Unable to get the nuclide index for Tally since "{0}" ' \
+                  'is not one of the nuclides'.format(nuclide)
+            raise KeyError(msg)
+        else:
+            return nuclide_index
+
+    def get_score_index(self, score):
+        """Returns the index in the Tally's results array for a score bin
+
+        Parameters
+        ----------
+        score : str
+            The score string (e.g., 'absorption', 'nu-fission')
+
+        Returns
+        -------
+        score_index : int
+            The index in the Tally data array for this score.
+
+        Raises
+        ------
+        ValueError
+            When the argument passed to the 'score' parameter cannot be found in
+            the Tally.
+
+        """
+
+        try:
+            score_index = self.scores.index(score)
+
+        except ValueError:
+            msg = 'Unable to get the score index for Tally since "{0}" ' \
```

Comment:
We should always include the downstream error in the new error as well. `'is not one of the scores: "{1}"'.format(score, err)`